confmap/provider/httpsprovider/provider.go

```diff
@@ -0,0 +1,103 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httpsprovider
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/confmap/provider/internal"
+)
+
+const (
+	schemeName = "https"
+)
+
+type provider struct{}
+
+// New returns a new confmap.Provider that reads the configuration from a file.
+//
+// This Provider supports "https" scheme, and can be called with a "uri" that follows:
+//   https-uri : https://host/xxx
+//
+// One example for https-uri be like: https://localhost:4444/getConfig
+//
+// Examples:
+// `https://localhost:4444/getConfig` - (unix, windows)
+func New() confmap.Provider {
+	return &provider{}
+}
+
+func (fmp *provider) Retrieve(_ context.Context, uri string, _ confmap.WatcherFunc) (confmap.Retrieved, error) {
+	if !strings.HasPrefix(uri, schemeName+"://") {
+		return confmap.Retrieved{}, fmt.Errorf("%q uri is not supported by %q provider", uri, schemeName)
+	}
+
+	// create a certificate pool, then add the root CA into it
+	myCAPath := os.Getenv("SSL_CERT_FILE")
+	if myCAPath == "" {
```

Comment:
This doesn't seem correct.  If no cert is provided then the system pool should be used.  In the normal case I'd expect that most people will not provide a custom CA or cert.

Reply:
Fixed, it now would not be treated as an error any more.

Reply:
The CR still seem incorrect. If myCAPath is empty, there is no point in trying to read or appending to the pool.